pkg/image/charts.go

```diff
@@ -0,0 +1,424 @@
+			for _, os := range strings.Split(osList, ",") {
+				switch strings.TrimSpace(strings.ToLower(os)) {
+				case "windows":
+					if osType == Windows {
+						addSourceToImage(imagesSet, imageName, chartNameAndVersion)
+						return
+					}
+				case "linux":
+					if osType == Linux {
+						addSourceToImage(imagesSet, imageName, chartNameAndVersion)
+						return
+					}
+				}
+			}
+		} else {
+			if inputMap["os"] != nil {
+				errors.Errorf("Field 'os:' for image %s contains neither a string nor nil", imageName)
+			}
+			if osType == Linux {
+				addSourceToImage(imagesSet, imageName, chartNameAndVersion)
+			}
+		}
+	})
+	return nil
+}
+
+// Walk a map and execute the given walk function for each node
+func walkMap(data interface{}, walkFunc func(map[interface{}]interface{})) {
+	if inputMap, isMap := data.(map[interface{}]interface{}); isMap {
+		// Run the walkFunc on the root node and each child node
+		walkFunc(inputMap)
+		for _, value := range inputMap {
+			walkMap(value, walkFunc)
+		}
+	} else if inputList, isList := data.([]interface{}); isList {
+		// Run the walkFunc on each element in the root node, ignoring the root itself
+		for _, elem := range inputList {
+			walkMap(elem, walkFunc)
+		}
+	}
+}
+
+// Convert min and max rancher version strings to a constraint string
+func minMaxToConstraintStr(min, max string) string {
+	if min != "" && max != "" {
+		return fmt.Sprintf("%s - %s", min, max)
+	}
+	if min != "" {
+		return fmt.Sprintf(">= %s", min)
+	}
+	if max != "" {
+		return fmt.Sprintf("<= %s", max)
+	}
+	return ""
+}
+
+// Check if the rancher version satisfies the given constraint range (E.g ">=2.5.0 <=2.6")
+func isRancherVersionInConstraintRange(rancherVersion, constraintStr string) (bool, error) {
+	if constraintStr == "" {
+		return false, errors.Errorf("Invalid constraint string: \"%s\"", constraintStr)
+	}
+	rancherSemVer, err := semver.NewVersion(rancherVersion)
+	if err != nil {
+		return false, err
+	}
+	// Removing the pre-release because the semver package will not consider a rancherVersion with a
+	// pre-releases unless the versions in the constraintStr has pre-releases as well.
+	// For example: rancherVersion "2.5.7-rc1" and constraint "2.5.6 - 2.5.8" will return false because
+	// there is no pre-release in the constraint "2.5.6 - 2.5.8" (This behavior is intentional).
+	rancherSemVerNoPreRelease, err := rancherSemVer.SetPrerelease("")
+	if err != nil {
+		return false, err
+	}
+	constraint, err := semver.NewConstraint(constraintStr)
+	if err != nil {
+		return false, err
+	}
+	return constraint.Check(&rancherSemVerNoPreRelease), nil
+}
+
+// Get the path to a chart's questions file if it has one
+func getQuestionsPath(versionLocalFiles []string) string {
+	for _, file := range versionLocalFiles {
+		basename := filepath.Base(file)
+		if basename == "questions.yaml" || basename == "questions.yml" {
+			return file
+		}
+	}
+	return ""
+}
+
+// Decode a questions file
```

Comment:
Use proper function definitions that start with function name. Applies to all functions with comment above. 
```suggestion
// decodeQuestions decodes a questions file.
```
Could probably explain more as well if you are going to include these. In this case renaming to `decodeQuestionsFile` would probably have the same effect.

Reply:
Ok, I'll update the decode functions to have more information.